packages/context/src/inject-config.ts

```diff
@@ -0,0 +1,182 @@
+ * expect(store2.optionX).to.eql(2);
+ * expect(store2.optionY).to.eql('b');
+ * ```
+ *
+ * @param configPath - Optional property path of the config. If is `''` or not
+ * present, the `config` object will be returned.
+ * @param metadata - Optional metadata to help the injection
+ */
+export function config(configPath?: string, metadata?: InjectionMetadata) {
+  configPath = configPath || '';
+  metadata = Object.assign(
+    {configPath, decorator: '@config', optional: true},
+    metadata,
+  );
+  return inject('', metadata, resolveFromConfig);
+}
+
+export namespace config {
+  export const getter = function injectConfigGetter(
+    configPath?: string,
+    metadata?: InjectionMetadata,
+  ) {
+    configPath = configPath || '';
+    metadata = Object.assign(
+      {configPath, decorator: '@config.getter', optional: true},
+      metadata,
+    );
+    return inject('', metadata, resolveAsGetterFromConfig);
+  };
+
+  export const view = function injectConfigView(
+    configPath?: string,
+    metadata?: InjectionMetadata,
+  ) {
+    configPath = configPath || '';
+    metadata = Object.assign(
+      {configPath, decorator: '@config.view', optional: true},
+      metadata,
+    );
+    return inject('', metadata, resolveAsViewFromConfig);
+  };
+}
+
+/**
+ * Get the key for the current binding on which dependency injection is
+ * performed
+ * @param session - Resolution session
+ */
+function getCurrentBindingKey(session: ResolutionSession) {
+  // The current binding is not set if `instantiateClass` is invoked directly
+  return session.currentBinding && session.currentBinding.key;
+}
+
+/**
+ * Resolver for `@config`
+ * @param ctx
+ * @param injection
+ * @param session
+ */
+function resolveFromConfig(
+  ctx: Context,
+  injection: Injection,
+  session: ResolutionSession,
+): ValueOrPromise<unknown> {
+  const bindingKey = getCurrentBindingKey(session);
+  // Return `undefined` if no current binding is present
+  if (!bindingKey) return undefined;
+  const meta = injection.metadata;
+  return ctx.getConfigAsValueOrPromise(bindingKey, meta.configPath, {
+    session,
+    optional: meta.optional,
+  });
+}
+
+/**
+ * Resolver from `@config.getter`
+ * @param ctx
+ * @param injection
+ * @param session
+ */
+function resolveAsGetterFromConfig(
+  ctx: Context,
+  injection: Injection,
+  session: ResolutionSession,
+) {
+  assertTargetType(injection, Function, 'Getter function');
+  const bindingKey = getCurrentBindingKey(session);
+  // We need to clone the session for the getter as it will be resolved later
+  const forkedSession = ResolutionSession.fork(session);
+  const meta = injection.metadata;
+  return async function getter() {
+    // Return `undefined` if no current binding is present
+    if (!bindingKey) return undefined;
+    return ctx.getConfigAsValueOrPromise(bindingKey, meta.configPath, {
+      session: forkedSession,
+      optional: meta.optional,
+    });
+  };
+}
+
+/**
+ * Resolver for `@config.view`
```

Comment:
I hope our TSDocs generator doesn't pick `@config` as some kind of keyword (from review POV it looks like one).

Reply:
It will not as it's quoted with `.